### PR TITLE
KCL-6857 Move configuration UI to new design

### PR DESCRIPTION
**Pull request by jankalfus**
Moved configuration UI to new design + did some code cleanup

### CODE REVIEW

**Comment by Simply007** [comment]
The CI probably fails because the pull request originated in different repository:
* https://github.com/PowerShell/PowerShell/issues/5911

**Review by PetrSvirak — CHANGES_REQUESTED** [review]
Thanks. The code looks mostly good to me. There few minor things to handle here or there. You plan some rebasing, so I will have yet another look at this PR afterward for sure.

File: DancingGoat/Areas/Admin/Views/Shared/_MessagesPartial.cshtml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
     const string imagePathPattern = "~/images/Admin/{0}.svg";
     var routeDataMessageBody = Context.Request.Query["MessageBody"].FirstOrDefault();
     var routeDataMessageType = Context.Request.Query["MessageType"].FirstOrDefault();
-    DancingGoat.Areas.Admin.Models.MessageModel viewBagMessage = null;
+    MessageModel viewBagMessage = null;
     string imageName = null;
     string imagePath = null;
 
-    if (!string.IsNullOrEmpty(ViewBag.Message))
+    if (ViewBag.Message != null)
```
[the remark this answers is not in the document]
Comment (jankalfus) [reply DancingGoat/Areas/Admin/Views/Shared/_MessagesPartial.cshtml:10]: Actually, this fixes a bug. The code exploded whenever it got to this point and the `Message` wasn't null. `ViewBag.Message` is of dynamic type, but in fact, it's set to an instance of `MessageModel`, not a `string`, hence can't be used as such, hence the explosion 😀

File: DancingGoat/wwwroot/css/UserMessage.less [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    padding: 2em;
+}
+
+.message-info {
+    background-color: @message-info-background;
+}
+
+.message-warning {
+    background-color: @message-warning-background;
+}
```
[the remark this answers is not in the document]
Comment (jankalfus) [reply DancingGoat/wwwroot/css/UserMessage.less:20]: Good point, I'll remove it.

File: DancingGoat/wwwroot/css/admin/UserMessage.less [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
-    padding: 2em;
+.hidden {
+  position: absolute;
+  width: 1px;
+  height: 1px;
+  padding: 0;
+  margin: -1px;
+  overflow: hidden;
+  clip: rect(0, 0, 0, 0);
+  white-space: nowrap; /* added line */
```
[the remark this answers is not in the document]
Comment (jankalfus) [reply DancingGoat/wwwroot/css/admin/UserMessage.less:14]: I don't either 😀 I'll remove it.

File: DancingGoat/wwwroot/css/admin/UserMessage.less [hunk 4]
```diff
@@ -1,34 +1,46 @@
...
+    padding: 0;
+  }
 }
 
 .message-error {
-    background-color: @message-error-background;
-}
+  & .user-message__img {
+    /* convert img from white to alert color */
+    filter: invert(100%) invert(8%) sepia(100%) saturate(7451%) hue-rotate(2deg) brightness(91%) contrast(108%);
```
[the remark this answers is not in the document]
Comment (jankalfus) [reply DancingGoat/wwwroot/css/admin/UserMessage.less:36]: There's a single image in _MessagesPartial.cshtml, and that file is used in both the admin part and dancing goat itself. What I guess we could do is have a div with background set to this image and then set the background in each respective less file. I'll check it out.

File: DancingGoat/Areas/Admin/Views/SelfConfig/Index.cshtml [hunk 5]
```diff
@@ -1,19 +1,18 @@
...
-    <div class="content">
-        <h1>@ViewBag.Title</h1>
-        <p>For your sample app to work, you should have a Kentico Kontent project containing content. Your app should be then configured with its project ID. You can either get it by signing in using your Kentico Kontent credentials or by signing up for a trial. Later, it will be converted to a free plan.</p>
+    <div class="kk-container">
+        <h1 class="headline-large">@ViewBag.Title</h1>
+        <p class="margin-top-xl">For your sample app to work, you should have a Kentico Kontent project containing content. Your app should be then configured with its project ID. You can either get it by signing in using your Kentico Kontent credentials or by signing up for a trial. Later, it will be converted to a free plan.</p>
     </div>
 </header>
-@{ await Html.RenderPartialAsync("~/Areas/Admin/Views/Shared/_MessagesPartial.cshtml");}
+@{ await Html.RenderPartialAsync("~/Areas/Admin/Views/Shared/_MessagesPartial.cshtml"); }
```
[the remark this answers is not in the document]
Comment (jankalfus) [reply DancingGoat/Areas/Admin/Views/SelfConfig/Index.cshtml:15]: I don't have access :(

[the remark this answers is not in the document]
Comment (jankalfus) [reply DancingGoat/Areas/Admin/Views/SelfConfig/Index.cshtml:15]: https://github.com/Kentico/kontent-sample-app-net/pull/125

File: DancingGoat/Areas/Admin/Views/Shared/_MessagesPartial.cshtml (hunk 1 above)
[the remark this answers is not in the document]
Comment (PetrSvirak) [reply DancingGoat/Areas/Admin/Views/Shared/_MessagesPartial.cshtml:10]: Lovely. You know the proverb: "Dynamic is the JavaScript of C#" 🤣

File: DancingGoat/wwwroot/css/.gitignore [hunk 6]
```diff
@@ -0,0 +1,2 @@
+*.css.map
+*.css
```
[the remark this answers is not in the document]
Comment (PetrSvirak) [reply DancingGoat/wwwroot/css/.gitignore:2]: Sorted out in https://github.com/Kentico/kontent-sample-app-net/pull/126

File: DancingGoat/wwwroot/css/admin/layout.less [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+@media (min-width: @screen-width-lg) {
+  .row-lg {
+    display: flex;
+  }
+
+  .row-lg--align-start {
+    align-items: flex-start;
+  }
+
+  .col + .col {
```
[the remark this answers is not in the document]
Comment (PetrSvirak) [reply DancingGoat/wwwroot/css/admin/layout.less:24]: I would hope Safari adoption sky-rockets with every release (in my imagination, mac update managers break in small tents in the queue in front of the servers, waiting for an official release for hours and days 👼), but I guess some users will get stuck with macOS 10.13 for quite some time yet 😢

File: DancingGoat/Areas/Admin/Views/SelfConfig/Index.cshtml (hunk 5 above)
[the remark this answers is not in the document]
Comment (PetrSvirak) [reply DancingGoat/Areas/Admin/Views/SelfConfig/Index.cshtml:15]: I believe we introduced the need here, so I would add the commit with the `.yml` to this PR, actually.

[the remark this answers is not in the document]
Comment (PetrSvirak) [reply DancingGoat/Areas/Admin/Views/SelfConfig/Index.cshtml:15]: Oh my: https://github.com/Kentico/kontent-sample-app-net/pull/126/checks?check_run_id=2682597130

[the remark this answers is not in the document]
Comment (PetrSvirak) [reply DancingGoat/Areas/Admin/Views/SelfConfig/Index.cshtml:15]: Resolved in https://github.com/Kentico/kontent-sample-app-net/pull/126

**Comment by jankalfus** [comment]
Closing in favor of https://github.com/Kentico/kontent-sample-app-net/pull/128

File: DancingGoat/wwwroot/css/UserMessage.less (hunk 2 above)
[the remark this answers is not in the document]
Comment (PetrSvirak) [reply DancingGoat/wwwroot/css/UserMessage.less:20]: Sorted out in https://github.com/Kentico/kontent-sample-app-net/pull/128

File: DancingGoat/wwwroot/css/admin/UserMessage.less (hunk 3 above)
[the remark this answers is not in the document]
Comment (PetrSvirak) [reply DancingGoat/wwwroot/css/admin/UserMessage.less:14]: Sorted out in https://github.com/Kentico/kontent-sample-app-net/pull/128

**Pull request closed without merging**